jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/SQLServerQueryRewriter.java

```diff
@@ -116,6 +116,14 @@ public String rewriteFilterItem(FilterItem item) {
                 final String dateTimeValue = "CAST('" + format.format(date) + "' AS DATETIME)";
 
                 sb.append(dateTimeValue);
+
+                //Remove TIMESTAMP token as SQL Server doesn't support it
+                int timestampIndex = sb.lastIndexOf("TIMESTAMP");
```

Comment:
It _is_ using `FilterItem.toSql()` directly when the operator is a string. It ends up in `AbstractQueryRewriter.rewriteFilterItem()` line 230 (in current master), which looks like this:
```Java
final String primaryFilterSql = item.toSql(isSchemaIncludedInColumnPaths());
```

If using `java.sql.Date` or `java.util.Date` instead of a string, everything works fine.